File: src/main/kotlin/com/fido/fcnetworksite/util/UserInfoUtils.java
package com.fido.fcnetworksite.util;

import com.fido.fcnetworksite.constant.PrefixConstant;
import com.fido.fcnetworksite.vo.UserVo;
import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpSession;

/**
 * @author: Fido Wang (fromwxf@gmail.com)
 * @date: 2018/12/17 21:35
 * @description:
 */
public class UserInfoUtils {

    public static UserVo getUserVo() {
        HttpServletRequest request =
                ((ServletRequestAttributes) (RequestContextHolder.currentRequestAttributes()))
                        .getRequest();
        HttpSession session = request.getSession();
        return (UserVo) session.getAttribute((PrefixConstant.SESSION_INFO_PREFIX));
    }

    public static void setUserVo(UserVo userVo) {
        HttpServletRequest request =
                ((ServletRequestAttributes) (RequestContextHolder.currentRequestAttributes()))
                        .getRequest();
        HttpSession session = request.getSession();
        session.setAttribute(PrefixConstant.SESSION_INFO_PREFIX, userVo);
    }
}

File: src/main/kotlin/com/fido/fcnetworksite/util/UserInfoUtils.kt
package com.fido.fcnetworksite.util

import com.fido.fcnetworksite.constant.PrefixConstant
import com.fido.fcnetworksite.dto.UserLoginDto
import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes

/**
 * @author: Fido Wang (fromwxf@gmail.com)
 * @date: 2018/12/16 02:03
 * @description:
 */
object UserInfoUtils {

    var userLoginDto: UserLoginDto
        get() {
            val request = (RequestContextHolder.currentRequestAttributes() as ServletRequestAttributes).request
            val session = request.session
            return session.getAttribute(PrefixConstant.SESSION_INFO_PREFIX) as UserLoginDto
        }
        set(userLoginDto) {
            val request = (RequestContextHolder.currentRequestAttributes() as ServletRequestAttributes).request
            val session = request.session
            session.setAttribute(PrefixConstant.SESSION_INFO_PREFIX, userLoginDto)
        }
}
